external_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 The LevelDB-Go and Pebble Authors. All rights reserved. Use
+// of this source code is governed by a BSD-style license that can be found in
+// the LICENSE file.
+
+package pebble_test
+
+import (
+	"bytes"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/pebble/metamorphic"
+	"github.com/cockroachdb/pebble/vfs"
+	"github.com/cockroachdb/pebble/vfs/errorfs"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/rand"
+)
+
+// TestIteratorErrors is a randomized test designed to ensure that errors
+// encountered by reads are properly propagated through to the user. It uses the
+// metamorphic tests configured with only write operations to first generate a
+// random database. It then uses the metamorphic tests to run a random set of
+// read operations against the generated database, randomly injecting errors at
+// the VFS layer. If an error is injected over the course of an operation, it
+// expects the error to surface to the operation output. If it doesn't, the test
+// fails.
+func TestIteratorErrors(t *testing.T) {
+	seed := time.Now().UnixNano()
```

Comment:
We should probably throw a `-seed` command line arg in this package too, just like `internal/metamorphic`.